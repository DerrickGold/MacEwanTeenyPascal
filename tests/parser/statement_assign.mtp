begin

	(*assignment statement: with factor (constant)*)
	something := 10;
	
	(*assignment statement: with factor (variable)*)
	something := something2;

	(*assignment statement: with factor (not factor)*)
	something := not something2;

	(*assignment statement: with terms*)
	something := 10 * 2;
	something := 10 div 2;
	something := 10 mod 2;
	something := 10 and 2;
	something := 10 shl 2;
	something := 10 shr 2;

	(*assignment statement: with simple expression unary*)
	something := -10;
	something := +10;

	(*assignment statement: binary adding operator*)
	something := 10 * 2 + 2 div 10;
	something := 10 * 2 - 2 div 10;
	something := 10 * 2 or 2 div 10;

	(*assignment statement: with expression*)
	something := (10 * 2 + 2 div 10 = 10 * 2 or 2 div 10);
	something := (10 * 2 + 2 div 10 <> 10 * 2 or 2 div 10);
	something := (10 * 2 + 2 div 10 < 10 * 2 or 2 div 10);
	something := (10 * 2 + 2 div 10 > 10 * 2 or 2 div 10);
	something := (10 * 2 + 2 div 10 <= 10 * 2 or 2 div 10);
	something := (10 * 2 + 2 div 10 >= 10 * 2 or 2 div 10);

	
end
.