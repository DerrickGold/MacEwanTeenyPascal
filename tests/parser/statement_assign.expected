[Head] (Block)
    [Child] (Statement List)
        [Child] (Assign Statement)
            [Child] 4, TOK_ID, something, (Simple Name)
            [Child] 4, TOK_NUM, 10, (Expression, Simple Expression, Factor, Constant)
        [Sibling] (Assign Statement)
            [Child] 7, TOK_ID, something, (Simple Name)
            [Child] 7, TOK_ID, something2, (Expression, Simple Expression, Factor, Constant, Simple Name)
        [Sibling] (Assign Statement)
            [Child] 10, TOK_ID, something, (Simple Name)
            [Child] 10, TOK_ID, something2, (Expression, Simple Expression, Factor, Constant, Simple Name, Not)
        [Sibling] (Assign Statement)
            [Child] 13, TOK_ID, something, (Simple Name)
            [Child] 13, TOK_STAR, *, (Expression, Simple Expression, Multiply Operator)
                [Child] 13, TOK_NUM, 10, (Factor, Constant, Term)
                [Child] 13, TOK_NUM, 2, (Factor, Constant, Term)
        [Sibling] (Assign Statement)
            [Child] 14, TOK_ID, something, (Simple Name)
            [Child] 14, TOK_KEY_DIV, div, (Expression, Simple Expression, Multiply Operator)
                [Child] 14, TOK_NUM, 10, (Factor, Constant, Term)
                [Child] 14, TOK_NUM, 2, (Factor, Constant, Term)
        [Sibling] (Assign Statement)
            [Child] 15, TOK_ID, something, (Simple Name)
            [Child] 15, TOK_KEY_MOD, mod, (Expression, Simple Expression, Multiply Operator)
                [Child] 15, TOK_NUM, 10, (Factor, Constant, Term)
                [Child] 15, TOK_NUM, 2, (Factor, Constant, Term)
        [Sibling] (Assign Statement)
            [Child] 16, TOK_ID, something, (Simple Name)
            [Child] 16, TOK_KEY_AND, and, (Expression, Simple Expression, Multiply Operator)
                [Child] 16, TOK_NUM, 10, (Factor, Constant, Term)
                [Child] 16, TOK_NUM, 2, (Factor, Constant, Term)
        [Sibling] (Assign Statement)
            [Child] 17, TOK_ID, something, (Simple Name)
            [Child] 17, TOK_KEY_SHL, shl, (Expression, Simple Expression, Multiply Operator)
                [Child] 17, TOK_NUM, 10, (Factor, Constant, Term)
                [Child] 17, TOK_NUM, 2, (Factor, Constant, Term)
        [Sibling] (Assign Statement)
            [Child] 18, TOK_ID, something, (Simple Name)
            [Child] 18, TOK_KEY_SHR, shr, (Expression, Simple Expression, Multiply Operator)
                [Child] 18, TOK_NUM, 10, (Factor, Constant, Term)
                [Child] 18, TOK_NUM, 2, (Factor, Constant, Term)
        [Sibling] (Assign Statement)
            [Child] 21, TOK_ID, something, (Simple Name)
            [Child] 21, TOK_MINUS, -, (Expression, Simple Expression, Unary Operator)
                [Child] 21, TOK_NUM, 10, (Factor, Constant)
        [Sibling] (Assign Statement)
            [Child] 22, TOK_ID, something, (Simple Name)
            [Child] 22, TOK_PLUS, +, (Expression, Simple Expression, Unary Operator)
                [Child] 22, TOK_NUM, 10, (Factor, Constant)
        [Sibling] (Assign Statement)
            [Child] 25, TOK_ID, something, (Simple Name)
            [Child] 25, TOK_PLUS, +, (Expression, Simple Expression, Binary Adding Operator)
                [Child] 25, TOK_STAR, *, (Simple Expression, Multiply Operator)
                    [Child] 25, TOK_NUM, 10, (Factor, Constant, Term)
                    [Child] 25, TOK_NUM, 2, (Factor, Constant, Term)
                [Child] 25, TOK_KEY_DIV, div, (Multiply Operator)
                    [Child] 25, TOK_NUM, 2, (Factor, Constant, Term)
                    [Child] 25, TOK_NUM, 10, (Factor, Constant, Term)
        [Sibling] (Assign Statement)
            [Child] 26, TOK_ID, something, (Simple Name)
            [Child] 26, TOK_MINUS, -, (Expression, Simple Expression, Binary Adding Operator)
                [Child] 26, TOK_STAR, *, (Simple Expression, Multiply Operator)
                    [Child] 26, TOK_NUM, 10, (Factor, Constant, Term)
                    [Child] 26, TOK_NUM, 2, (Factor, Constant, Term)
                [Child] 26, TOK_KEY_DIV, div, (Multiply Operator)
                    [Child] 26, TOK_NUM, 2, (Factor, Constant, Term)
                    [Child] 26, TOK_NUM, 10, (Factor, Constant, Term)
        [Sibling] (Assign Statement)
            [Child] 27, TOK_ID, something, (Simple Name)
            [Child] 27, TOK_KEY_OR, or, (Expression, Simple Expression, Binary Adding Operator)
                [Child] 27, TOK_STAR, *, (Simple Expression, Multiply Operator)
                    [Child] 27, TOK_NUM, 10, (Factor, Constant, Term)
                    [Child] 27, TOK_NUM, 2, (Factor, Constant, Term)
                [Child] 27, TOK_KEY_DIV, div, (Multiply Operator)
                    [Child] 27, TOK_NUM, 2, (Factor, Constant, Term)
                    [Child] 27, TOK_NUM, 10, (Factor, Constant, Term)
        [Sibling] (Assign Statement)
            [Child] 30, TOK_ID, something, (Simple Name)
            [Child] 30, TOK_EQ, =, (Expression, Simple Expression, Factor, Relational Operator)
                [Child] 30, TOK_PLUS, +, (Simple Expression, Binary Adding Operator)
                    [Child] 30, TOK_STAR, *, (Simple Expression, Multiply Operator)
                        [Child] 30, TOK_NUM, 10, (Factor, Constant, Term)
                        [Child] 30, TOK_NUM, 2, (Factor, Constant, Term)
                    [Child] 30, TOK_KEY_DIV, div, (Multiply Operator)
                        [Child] 30, TOK_NUM, 2, (Factor, Constant, Term)
                        [Child] 30, TOK_NUM, 10, (Factor, Constant, Term)
                [Child] 30, TOK_KEY_OR, or, (Simple Expression, Binary Adding Operator)
                    [Child] 30, TOK_STAR, *, (Simple Expression, Multiply Operator)
                        [Child] 30, TOK_NUM, 10, (Factor, Constant, Term)
                        [Child] 30, TOK_NUM, 2, (Factor, Constant, Term)
                    [Child] 30, TOK_KEY_DIV, div, (Multiply Operator)
                        [Child] 30, TOK_NUM, 2, (Factor, Constant, Term)
                        [Child] 30, TOK_NUM, 10, (Factor, Constant, Term)
        [Sibling] (Assign Statement)
            [Child] 31, TOK_ID, something, (Simple Name)
            [Child] 31, TOK_NOTEQ, <>, (Expression, Simple Expression, Factor, Relational Operator)
                [Child] 31, TOK_PLUS, +, (Simple Expression, Binary Adding Operator)
                    [Child] 31, TOK_STAR, *, (Simple Expression, Multiply Operator)
                        [Child] 31, TOK_NUM, 10, (Factor, Constant, Term)
                        [Child] 31, TOK_NUM, 2, (Factor, Constant, Term)
                    [Child] 31, TOK_KEY_DIV, div, (Multiply Operator)
                        [Child] 31, TOK_NUM, 2, (Factor, Constant, Term)
                        [Child] 31, TOK_NUM, 10, (Factor, Constant, Term)
                [Child] 31, TOK_KEY_OR, or, (Simple Expression, Binary Adding Operator)
                    [Child] 31, TOK_STAR, *, (Simple Expression, Multiply Operator)
                        [Child] 31, TOK_NUM, 10, (Factor, Constant, Term)
                        [Child] 31, TOK_NUM, 2, (Factor, Constant, Term)
                    [Child] 31, TOK_KEY_DIV, div, (Multiply Operator)
                        [Child] 31, TOK_NUM, 2, (Factor, Constant, Term)
                        [Child] 31, TOK_NUM, 10, (Factor, Constant, Term)
        [Sibling] (Assign Statement)
            [Child] 32, TOK_ID, something, (Simple Name)
            [Child] 32, TOK_LESS, <, (Expression, Simple Expression, Factor, Relational Operator)
                [Child] 32, TOK_PLUS, +, (Simple Expression, Binary Adding Operator)
                    [Child] 32, TOK_STAR, *, (Simple Expression, Multiply Operator)
                        [Child] 32, TOK_NUM, 10, (Factor, Constant, Term)
                        [Child] 32, TOK_NUM, 2, (Factor, Constant, Term)
                    [Child] 32, TOK_KEY_DIV, div, (Multiply Operator)
                        [Child] 32, TOK_NUM, 2, (Factor, Constant, Term)
                        [Child] 32, TOK_NUM, 10, (Factor, Constant, Term)
                [Child] 32, TOK_KEY_OR, or, (Simple Expression, Binary Adding Operator)
                    [Child] 32, TOK_STAR, *, (Simple Expression, Multiply Operator)
                        [Child] 32, TOK_NUM, 10, (Factor, Constant, Term)
                        [Child] 32, TOK_NUM, 2, (Factor, Constant, Term)
                    [Child] 32, TOK_KEY_DIV, div, (Multiply Operator)
                        [Child] 32, TOK_NUM, 2, (Factor, Constant, Term)
                        [Child] 32, TOK_NUM, 10, (Factor, Constant, Term)
        [Sibling] (Assign Statement)
            [Child] 33, TOK_ID, something, (Simple Name)
            [Child] 33, TOK_GREATER, >, (Expression, Simple Expression, Factor, Relational Operator)
                [Child] 33, TOK_PLUS, +, (Simple Expression, Binary Adding Operator)
                    [Child] 33, TOK_STAR, *, (Simple Expression, Multiply Operator)
                        [Child] 33, TOK_NUM, 10, (Factor, Constant, Term)
                        [Child] 33, TOK_NUM, 2, (Factor, Constant, Term)
                    [Child] 33, TOK_KEY_DIV, div, (Multiply Operator)
                        [Child] 33, TOK_NUM, 2, (Factor, Constant, Term)
                        [Child] 33, TOK_NUM, 10, (Factor, Constant, Term)
                [Child] 33, TOK_KEY_OR, or, (Simple Expression, Binary Adding Operator)
                    [Child] 33, TOK_STAR, *, (Simple Expression, Multiply Operator)
                        [Child] 33, TOK_NUM, 10, (Factor, Constant, Term)
                        [Child] 33, TOK_NUM, 2, (Factor, Constant, Term)
                    [Child] 33, TOK_KEY_DIV, div, (Multiply Operator)
                        [Child] 33, TOK_NUM, 2, (Factor, Constant, Term)
                        [Child] 33, TOK_NUM, 10, (Factor, Constant, Term)
        [Sibling] (Assign Statement)
            [Child] 34, TOK_ID, something, (Simple Name)
            [Child] 34, TOK_LTEQ, <=, (Expression, Simple Expression, Factor, Relational Operator)
                [Child] 34, TOK_PLUS, +, (Simple Expression, Binary Adding Operator)
                    [Child] 34, TOK_STAR, *, (Simple Expression, Multiply Operator)
                        [Child] 34, TOK_NUM, 10, (Factor, Constant, Term)
                        [Child] 34, TOK_NUM, 2, (Factor, Constant, Term)
                    [Child] 34, TOK_KEY_DIV, div, (Multiply Operator)
                        [Child] 34, TOK_NUM, 2, (Factor, Constant, Term)
                        [Child] 34, TOK_NUM, 10, (Factor, Constant, Term)
                [Child] 34, TOK_KEY_OR, or, (Simple Expression, Binary Adding Operator)
                    [Child] 34, TOK_STAR, *, (Simple Expression, Multiply Operator)
                        [Child] 34, TOK_NUM, 10, (Factor, Constant, Term)
                        [Child] 34, TOK_NUM, 2, (Factor, Constant, Term)
                    [Child] 34, TOK_KEY_DIV, div, (Multiply Operator)
                        [Child] 34, TOK_NUM, 2, (Factor, Constant, Term)
                        [Child] 34, TOK_NUM, 10, (Factor, Constant, Term)
        [Sibling] (Assign Statement)
            [Child] 35, TOK_ID, something, (Simple Name)
            [Child] 35, TOK_GTEQ, >=, (Expression, Simple Expression, Factor, Relational Operator)
                [Child] 35, TOK_PLUS, +, (Simple Expression, Binary Adding Operator)
                    [Child] 35, TOK_STAR, *, (Simple Expression, Multiply Operator)
                        [Child] 35, TOK_NUM, 10, (Factor, Constant, Term)
                        [Child] 35, TOK_NUM, 2, (Factor, Constant, Term)
                    [Child] 35, TOK_KEY_DIV, div, (Multiply Operator)
                        [Child] 35, TOK_NUM, 2, (Factor, Constant, Term)
                        [Child] 35, TOK_NUM, 10, (Factor, Constant, Term)
                [Child] 35, TOK_KEY_OR, or, (Simple Expression, Binary Adding Operator)
                    [Child] 35, TOK_STAR, *, (Simple Expression, Multiply Operator)
                        [Child] 35, TOK_NUM, 10, (Factor, Constant, Term)
                        [Child] 35, TOK_NUM, 2, (Factor, Constant, Term)
                    [Child] 35, TOK_KEY_DIV, div, (Multiply Operator)
                        [Child] 35, TOK_NUM, 2, (Factor, Constant, Term)
                        [Child] 35, TOK_NUM, 10, (Factor, Constant, Term)
        [Sibling] (Null Statement)
